File: src/app/services/appointment.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Appointment } from '../models/appointment.model';

@Injectable({
  providedIn: 'root'
})
export class AppointmentService {
 
  //baseUrl:string="https://ide-bacdaddceafccefebdfdedddbdfbedbebb.premiumproject.examly.io/proxy/8080/api/appointments"

  baseUrl:string="http://localhost:8080/api/appointments"


  constructor(private httpClient:HttpClient) { }

  public getAppointments():Observable<Appointment[]>{
    return this.httpClient.get(this.baseUrl) as Observable<Appointment[]>
  }


  public getAppointmentsById(id:number):Observable<Appointment>{
    return this.httpClient.get(this.baseUrl+'/'+id) as Observable<Appointment>
  }

  public getAppointmentsByUserId(id:number):Observable<Appointment[]>{
    return this.httpClient.get(this.baseUrl+'/user/'+id) as Observable<Appointment[]>
  }

  public createAppointments(appointment:Appointment):Observable<Appointment>{
    return this.httpClient.post(this.baseUrl,appointment) as Observable<Appointment>
  }

  public updateAppointment(id:number,appointment:Appointment):Observable<Appointment>{
    return this.httpClient.put(this.baseUrl+'/'+id,appointment) as Observable<Appointment>
  }

  public deleteAppointment(id:number):Observable<any>{
    return this.httpClient.delete(this.baseUrl+'/'+id) as Observable<any>
  }

  updateAppointmentStatus(id: number, status: string): Observable<Appointment> {
    return this.httpClient.put<Appointment>(this.baseUrl+'/'+id, status);
  }


}

File: springapp/src/main/java/com/examly/springapp/service/AppointmentService.java
package com.examly.springapp.service;

import java.util.List;
import java.util.Optional;


import com.examly.springapp.model.Appointment;


public interface AppointmentService {
    public List<Appointment>  getAllAppointements();
    public Appointment addAppointmnet(Appointment appointment);
    public Appointment getAppointmentById(int appointmentId);
    public Appointment updateAppointment(int appointmentId,Appointment appointment);
    public void deleteAppointment(int appointmentId);
    public List<Appointment> getAppointmentByUserId(int userId);
    public Appointment updateAppointmentStatus(int appointmentId, Appointment status);
}

